nomad/structs/structs.go

```diff
@@ -11041,20 +11125,22 @@ func (a *Allocation) SupportsDisconnectedClients(serverSupportsDisconnectedClien
 	if a.Job != nil {
 		tg := a.Job.LookupTaskGroup(a.TaskGroup)
 		if tg != nil {
-			return tg.MaxClientDisconnect != nil
+			return tg.GetDisconnectLostTimeout() != 0
 		}
 	}
 
 	return false
 }
 
 // PreventRescheduleOnLost determines if an alloc allows to have a replacement
-// when lost.
-func (a *Allocation) PreventRescheduleOnLost() bool {
+// when Disconnected.
+func (a *Allocation) PreventRescheduleOnDisconnect() bool {
```

Comment:
Strictly speaking `lost` is a valid client status for an alloc, while `disconnect` is not, so this method is checking if an alloc should not reschedule if it were to become `lost`. 

But also the alloc would be `unknown`, not `lost`, so the new name is probably fine 😅